public/app/plugins/datasource/azuremonitor/components/AzureCheatSheet.tsx

```diff
@@ -0,0 +1,297 @@
+import { css } from '@emotion/css';
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { getBackendSrv, reportInteraction } from '@grafana/runtime';
+import {
+  Button,
+  Card,
+  Collapse,
+  CustomScrollbar,
+  InlineField,
+  Input,
+  LoadingPlaceholder,
+  Select,
+  useStyles2,
+} from '@grafana/ui';
+
+import tokenizer from '../../cloudwatch/language/cloudwatch-logs/syntax';
+import { RawQuery } from '../../prometheus/querybuilder/shared/RawQuery';
+import { AzureMonitorQuery, AzureQueryType } from '../types';
+
+type Category = {
+  displayName: string;
+  id: string;
+  related: {
+    queries: string[];
+    tables: string[];
+  };
+};
+
+type CheatsheetQuery = {
+  body: string;
+  description: string;
+  displayName: string;
+  id: string;
+  properties: {
+    ExampleQuery: boolean;
+    QueryAttributes: {
+      isMultiResource: boolean;
+    };
+  };
+  related: {
+    categories: string[];
+    resourceTypes: string[];
+    tables: string[];
+  };
+  tags: {
+    Topic: string[];
+  };
+};
+
+type CheatsheetQueries = {
+  [key: string]: CheatsheetQuery[];
+};
+
+type DropdownCategories = {
+  [key: string]: boolean;
+};
+
+type Props = {
+  onClickExample: (query: AzureMonitorQuery) => void;
+  query: AzureMonitorQuery;
+};
+
+const AzureCheatSheet = (props: Props) => {
+  const [cheatsheetQueries, setCheatsheetQueries] = useState<CheatsheetQueries | null>(null);
+  const [areDropdownsOpen, setAreDropdownsOpen] = useState<DropdownCategories>({});
+  const [visibleQueries, setVisibleQueries] = useState<CheatsheetQueries | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [searchInputValue, setSearchInputValue] = useState('');
+  const styles = useStyles2(getStyles);
+  const lang = { grammar: tokenizer, name: 'kql' };
+  const dropdownMenu = useMemo(() => {
+    if (cheatsheetQueries) {
+      return Object.keys(cheatsheetQueries).map((category): SelectableValue<string> => {
+        return {
+          label: category,
+          value: category,
+        };
+      });
+    }
+    return [];
+  }, [cheatsheetQueries]);
+
+  const getCheatsheetQueries = async () => {
```

Comment:
Can we move this function up to the datasource and then add the request to the backend datasource?